lib/popcorn/pop_repos.nit

```diff
@@ -331,6 +331,57 @@ class MongoRepository[E: Serializable]
 	end
 end
 
+# Base serializable entity that can go into a JsonRepository
+#
+# Provide boiler plate implementation of all object serializable to json.
+#
+# `id` is used as a primary key for `find_by_id`.
+#
+# Subclassing RepoObject makes it easy to create a serializable class:
+# ~~~
+# import popcorn::pop_repos
+#
+# class Album
+#	super RepoObject
+#	serialize
+#
+#	var title: String
+#	var price: Float
+# end
+# ~~~
+#
+# Do not forget the `serialize` annotation else the fields will not be serialized.
+#
+# It is also possible to redefine the `id` primary key to use your own:
+# ~~~
+# import popcorn::pop_repos
+#
+# class Order
+#	super RepoObject
+#	serialize
+#
+#	redef var id = "order-{get_time}"
+#
+#	# ...
+#
+# end
+# ~~~
+abstract class RepoObject
+	super Jsonable
+	serialize
+
+	# `self` unique id.
+	#
+	# This attribute is serialized under the key `_id` to be used
+	# as primary key by MongoDb
+	var id: String = (new MongoObjectId).id is serialize_as "_id"
+
+	redef fun to_s do return id
+	redef fun ==(o) do return o isa SELF and id == o.id
```

Comment:
I just wanted that you document what is the specific semantic of `==` here. Saying (implicitly) that `==` is redefined is not enough unless you explain what the new semantic is.
Also, I do not know if it is expected that subclasses of `RepoObject` should define their own specific `==`, or if it is a good idea, or if it is a bad idea.
Things can become quite complex since serialization and mangodb are also involved.
I'm quite annoying here because `==` (and `hash`) are quite ubiquitously in core data-structures like Hash or Array.